updateQDKVersion.ps1

```diff
@@ -0,0 +1,61 @@
+$ver = $args[0]
+Write-Host $ver
+
+$csString = 'PackageReference Include=\"Microsoft\.Quantum\.[a-zA-Z\.]+\" Version=\"(?<oldVersion>[0-9|\.]+)\"'
+$csFiles = (Select-String -Path "**\*.csproj" -pattern "Microsoft.Quantum" | Select-Object -Unique Path)
+$csFiles | ForEach-Object {
+    (Get-Content -Encoding UTF8 $_.Path) | ForEach-Object {
+         $isQuantumPackage = $_ -match $csString
+         if ($isQuantumPackage) {
+             $_ -replace $Matches.oldVersion, $ver
+         } else {
+             $_
+         }
+    } | Set-Content -Encoding UTF8 $_.Path
```

Comment:
For compatibility reasons, this line writes out using an old nonstandard variant of UTF8 that has a byte-order mark (BOM) at the beginning. On PowerShell 6 and later, `-Encoding UTF8NoBom` can be used to suppress the BOM. We've before had cases where some tools fail to recognize UTF8-encoded files with the BOM, unfortunately.

At this point, PowerShell 5.1 and earlier (also known as Windows PowerShell) are fairly old, such that I think we can use `UTF8NoBom` fairly safely. To be sure, you can include a `#Requires -PSEdition Core` statement at the top of the script to make sure that the `UTF8NoBom` value is supported.

Reply:
Sorry, I missed this comment in the first read. I don't think 5.1 is _that_ old - I set up my laptop less than a year ago and it has 5.1 - so I don't think we want to phase it out completely. I kept UTF8 here and removed requirement for Core edition.